native_client/swift/deepspeech_ios/deepspeech.h

```diff
@@ -0,0 +1,357 @@
+ *         responsible for freeing Metadata by calling {@link DS_FreeMetadata()}.
+ *         Returns NULL on error.
+ */
+DEEPSPEECH_EXPORT
+Metadata* DS_IntermediateDecodeWithMetadata(const StreamingState* aSctx,
+                                            unsigned int aNumResults);
+
+/**
+ * @brief Compute the final decoding of an ongoing streaming inference and return
+ *        the result. Signals the end of an ongoing streaming inference.
+ *
+ * @param aSctx A streaming state pointer returned by {@link DS_CreateStream()}.
+ *
+ * @return The STT result. The user is responsible for freeing the string using
+ *         {@link DS_FreeString()}.
+ *
+ * @note This method will free the state pointer (@p aSctx).
+ */
+DEEPSPEECH_EXPORT
+char* DS_FinishStream(StreamingState* aSctx);
+
+/**
+ * @brief Compute the final decoding of an ongoing streaming inference and return
+ *        results including metadata. Signals the end of an ongoing streaming
+ *        inference.
+ *
+ * @param aSctx A streaming state pointer returned by {@link DS_CreateStream()}.
+ * @param aNumResults The number of candidate transcripts to return.
+ *
+ * @return Metadata struct containing multiple candidate transcripts. Each transcript
+ *         has per-token metadata including timing information. The user is
+ *         responsible for freeing Metadata by calling {@link DS_FreeMetadata()}.
+ *         Returns NULL on error.
+ *
+ * @note This method will free the state pointer (@p aSctx).
+ */
+DEEPSPEECH_EXPORT
+Metadata* DS_FinishStreamWithMetadata(StreamingState* aSctx,
+                                      unsigned int aNumResults);
+
+/**
+ * @brief Destroy a streaming state without decoding the computed logits. This
+ *        can be used if you no longer need the result of an ongoing streaming
+ *        inference and don't want to perform a costly decode operation.
+ *
+ * @param aSctx A streaming state pointer returned by {@link DS_CreateStream()}.
+ *
+ * @note This method will free the state pointer (@p aSctx).
+ */
+DEEPSPEECH_EXPORT
+void DS_FreeStream(StreamingState* aSctx);
+
+/**
+ * @brief Free memory allocated for metadata information.
+ */
+DEEPSPEECH_EXPORT
+void DS_FreeMetadata(Metadata* m);
+
+/**
+ * @brief Free a char* string returned by the DeepSpeech API.
+ */
+DEEPSPEECH_EXPORT
+void DS_FreeString(char* str);
+
+/**
+ * @brief Returns the version of this library. The returned version is a semantic
+ *        version (SemVer 2.0.0). The string returned must be freed with {@link DS_FreeString()}.
+ *
+ * @return The version string.
+ */
+DEEPSPEECH_EXPORT
+char* DS_Version();
+
+/**
+ * @brief Returns a textual description corresponding to an error code.
+ *        The string returned must be freed with @{link DS_FreeString()}.
+ *
+ * @return The error description.
+ */
+DEEPSPEECH_EXPORT
+char* DS_ErrorCodeToErrorMessage(int aErrorCode);
+
+#undef DEEPSPEECH_EXPORT
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* DEEPSPEECH_H */
```

Comment:
but I fail to find a difference. i'm not a big fan of copies around of headers like that, it feels like it's going to blow up some time or another, can't we symlink (it's osx so no reason it would not work)?

Reply:
See above, it's supposed to be just a reference, it should have been included here.